src/history/hs_rec.c

```diff
@@ -225,9 +225,15 @@ __hs_insert_record_with_btree(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BT
                 WT_ERR(cursor->get_value(cursor, &hs_stop_durable_ts_diag, &durable_timestamp_diag,
                   &upd_type_full_diag, existing_val));
                 WT_ERR(__wt_compare(session, NULL, existing_val, hs_value, &cmp));
+                /*
+                 * Check if the existing HS value is same as the new value we are about to insert.
+                 * We can skip this check if the existing value has a globally visible stop time.
+                 */
```

Comment:
Maybe explain what a globally visible stop time implies?

Reply:
Added an addition sentence in 5cf76b8.